block.go

```diff
@@ -179,6 +179,10 @@ type BlockMetaCompaction struct {
 	Level int `json:"level"`
 	// ULIDs of all source head blocks that went into the block.
 	Sources []ulid.ULID `json:"sources,omitempty"`
+	// As we dont write empty blocks, we need this to mark
+	// if the block is deletable if during compaction this block
+	// resulted in an empty block.
```

Comment:
// Indicates that during compaction it resulted in a block without any samples 
// so it should be deleted on the next reload.